application_test.go

```diff
@@ -198,14 +198,32 @@ func TestApplicationEnvs(t *testing.T) {
 	app := NewDockerApplication()
 	assert.Nil(t, app.Env)
 
-	app.AddEnv("hello", "world").AddEnv("foo", "bar")
-	assert.Equal(t, 2, len(*app.Env))
-	assert.Equal(t, "world", (*app.Env)["hello"])
-	assert.Equal(t, "bar", (*app.Env)["foo"])
+	app.AddEnv("hello", "world").AddEnv("foo", "bar").AddSecret("top", "secret1", "/path/to/secret")
+	assert.Equal(t, 2, len(app.Env))
+	assert.Equal(t, 1, len(app.Secrets))
+	assert.Equal(t, "world", app.Env["hello"])
+	assert.Equal(t, "bar", app.Env["foo"])
+	assert.Equal(t, "/path/to/secret", app.Secrets["secret1"].Source)
+	assert.Equal(t, "top", app.Secrets["secret1"].EnvVar)
 
 	app.EmptyEnvs()
 	assert.NotNil(t, app.Env)
```

Comment:
If `app.Env` is a non-pointer, it can never be nil.

In case we decide to redeclare a pointer on `app.Env`: Could you please wrap the assert in an if statement and put the following statement on line 211 inside the if block? That is,

```go
if assert.NotNil(t, app.Env) {
  assert.Equal(t, 0, len(app.Env))
}
```